server/controllers/finance/accounts/references.compute.js

```diff
@@ -0,0 +1,184 @@
+      glb.fiscalYear = fiscalYear;
+
+      return db.one(queryAccountReference, [abbr, isAmoDep]);
+    })
+    .then(ar => {
+      return getValueForReference(
+        ar.abbr,
+        ar.isAmoDep,
+        ar.description,
+        glb.fiscalYear.period_number,
+        glb.fiscalYear.id
+      );
+    })
+    .then(data => {
+      return data[0];
+    });
+}
+
+/**
+ * @method getValueForReference
+ *
+ * @description
+ * Returns computed value of the reference in a given period and fiscal_year
+ *
+ * @param {number} fiscalYearId
+ * @param {number} periodNumber
+ * @param {string} abbr - the reference of accounts. ex. AA or AX
+ * @param {boolean} isAmoDep - the concerned reference is for amortissement, depreciation or provision
+ */
+function getValueForReference(abbr, isAmoDep = 0, referenceDescription, periodNumber, fiscalYearId) {
+  const queryTotals = `
+  SELECT abbr, is_amo_dep, description,
+    IFNULL(debit, 0) AS debit, IFNULL(credit, 0) AS credit, IFNULL(balance, 0) AS balance FROM (
+      SELECT ? AS abbr, ? AS is_amo_dep, ? AS description,
+        SUM(IFNULL(pt.debit, 0)) AS debit, SUM(IFNULL(pt.credit, 0)) AS credit,
+        SUM(IFNULL(pt.debit - pt.credit, 0)) AS balance
+      FROM period_total pt 
+      JOIN period p ON p.id = pt.period_id
+      WHERE pt.fiscal_year_id = ? AND pt.locked = 0 AND p.number BETWEEN 0 AND ? AND pt.account_id IN (?)
+    )z
+  `;
+
+  return getAccountsForReference(abbr, isAmoDep)
+    .then(accounts => {
+      const accountIds = accounts.map(a => a.account_id);
+      const parameters = [
+        abbr,
+        isAmoDep,
+        referenceDescription,
+        fiscalYearId,
+        periodNumber,
+        accountIds.length ? accountIds : null,
+      ];
+      return db.exec(queryTotals, parameters);
+    });
+}
+
+/**
+ * @method getAccountsForReference
+ *
+ * @description
+ * Returns all accounts concerned by the reference without exception accounts
+ *
+ * @param {string} abbr - the reference of accounts. ex. AA or AX
+ * @param {boolean} isAmoDep - the concerned reference is for amortissement, depreciation or provision
+ */
+function getAccountsForReference(abbr, isAmoDep = 0) {
+  /**
+   * Get the list of accounts of the reference without excepted accounts
+   * mysql implementation of minus operator
+   * @link http://www.mysqltutorial.org/mysql-minus/
+   */
+  const queryAccounts = `
+    SELECT includeTable.account_id, includeTable.account_number FROM (
+      SELECT DISTINCT 
+        account.id AS account_id, account.number AS account_number, t.is_exception FROM account
+        JOIN (
+          SELECT a.id, a.number, ar.is_amo_dep, ari.is_exception FROM account a 
+          JOIN account_reference_item ari ON ari.account_id = a.id
+          JOIN account_reference ar ON ar.id = ari.account_reference_id
+          WHERE ar.abbr LIKE ? AND ar.is_amo_dep = ? AND ari.is_exception = 0
+        ) AS t ON LEFT(account.number, CHAR_LENGTH(t.number)) LIKE t.number 
```

Comment:
Why are we joining on `LEFT account.number`?  Can't we make this an account type?

Reply:
Because we want to have all accounts from the account table starting with account number of accounts in account_reference_item table

Reply:
Can we use `=` instead of `LIKE`?

Instead of linking the account number, would it be possible to use the `account_id`?  Then we could have a foreign key relationship which would be more reliable than parsing numbers, I think.